fabric-particles-v1/src/main/java/net/fabricmc/fabric/impl/client/particle/FabricParticleManager.java

```diff
@@ -81,19 +90,38 @@ public boolean loadParticle(ResourceManager manager, Identifier id) {
 				throw new IllegalStateException("(Fabric) Missing texture list for particle " + id);
 			}
 
-			provider.setSprites(spriteIds);
+			unloadedSprites.put(id, spriteIds.stream()
+					.map(sprite -> new Identifier(sprite.getNamespace(), "particle/" + sprite.getPath()))
```

Comment:
This adding of "particle/" seems to be breaking @NinjaPhenix 's mod.

```
[23:37:58] [Server-Worker-5/ERROR]: Using missing texture, unable to load torcherino:textures/particle/particle/flame.png : java.io.FileNotFoundException: torcherino:textures/particle/particle/flame.png
[23:37:58] [Server-Worker-7/ERROR]: Using missing texture, unable to load torcherino:textures/particle/particle/compressed_flame.png : java.io.FileNotFoundException: torcherino:textures/particle/particle/compressed_flame.png
[23:37:58] [Server-Worker-7/ERROR]: Using missing texture, unable to load torcherino:textures/particle/particle/double_compressed_flame.png : java.io.FileNotFoundException: torcherino:textures/particle/particle/double_compressed_flame.png
```

![](https://cdn.discordapp.com/attachments/566276937035546624/689256296368570555/unknown.png)

Reply:
vanilla automatically appends the filepath in the json, this change is aimed at keeping the formatting consistent
![image](https://user-images.githubusercontent.com/11313954/76808716-a4587680-67f9-11ea-84d5-ac95353afd4e.png)

Reply:
Humm, ok. Breaking mods isnt something that we usually like to do. This will need further input.

Reply:
to add to this, some of the modders I've contacted while trying to get this to work have been confused by the fact you have to specify the `particles/` filepath in modded particle jsons, so this change is probably for the best however "bad" it might seem

Reply:
From what I can tell I've likely implemented it badly in torcherino.
Also looking on github only two mods would break. So I really don't mind too much, would only take me a few minutes to update and release a new build.

Reply:
From what I can tell, I'm not actually sure this is putting particles into the particle atlas. The best solution would be to have a way to automatically add particle textures from mods into the particle atlas based on folders, but that would still break textures. I think breaking textures is a preferable alternative to the difficulties currently present in custom particle textures, though, and since it's a find-and-replace fix in a JSON file it's a change that's worth it to make.

Reply:
if you're referring to the contents of the pr, the change that this comment chain sprung from "collects" textures from deserialized jsons into a map that's then baked into the final particle atlas :^)

Reply:
ahhh, ok, it was hard to fully understand what the code was doing